middle_end/flambda2/simplify/simplify_let_cont_expr.ml

```diff
@@ -1235,17 +1459,22 @@ let after_downwards_traversal_of_body ~simplify_expr
             let cont_uses_env = CUE.remove cont_uses_env_so_far cont in
             DA.with_continuation_uses_env dacc ~cont_uses_env
           in
-          let data : after_downwards_traversal_of_body_and_handlers_data =
+          let dacc =
+            DA.with_are_lifting_conts dacc previous_are_lifting_conts
+          in
+          let data : after_downwards_traversal_of_body_and_handlers_rebuild =
             { rebuild_body;
               cont_uses_env = cont_uses_env_so_far;
+              lifted_params;
               invariant_params = Bound_parameters.empty;
               invariant_extra_params_and_args = EPA.empty;
               handlers = Continuation.Map.singleton cont rebuild;
               at_unit_toplevel
             }
           in
-          after_downwards_traversal_of_body_and_handlers data ~down_to_up dacc))
-  | Recursive { continuation_handlers; invariant_params } ->
+          k dacc data))
+  | Recursive { continuation_handlers; invariant_params; lifted_params } ->
+    let dacc = DA.with_are_lifting_conts dacc Are_lifting_conts.no_lifting in
```

Comment:
This would need a comment to justify that we don't lift inside a recursive continuation.

Reply:
And maybe this would be a good way to test some seldomly used other codes. ;)

Reply:
comment added